src/charm.py

```diff
@@ -740,6 +721,144 @@ def _gen_wp_config(self):
         )
         return "\n".join(wp_config)
 
+    def _container(self):
+        return self.unit.get_container("wordpress")
+
+    def _wordpress_service_exists(self):
+        return "wordpress" in self._container().get_plan().services
+
+    def _stop_server(self):
+        """Stop WordPress (apache) server
+
+        This operation is idempotence.
+        """
+        if self._wordpress_service_exists():
+            self._container().stop("wordpress")
+
+    def _wp_is_installed(self):
+        """Check if WordPress is installed (check if WordPress related tables exist in database)"""
+        process = self._container().exec(["wp", "core", "is-installed"])
+        try:
+            process.wait()
+            return True
+        except ops.pebble.ExecError:
+            return False
+
+    def _wp_install_cmd(self):
+        """Generate wp-cli command used to install WordPress on database"""
+        initial_settings = yaml.safe_load(self.model.config["initial_settings"])
+        admin_user = initial_settings.get("user_name", "admin_username")
+        admin_email = initial_settings.get("admin_email", "name@example.com")
+        default_admin_password = self._replica_relation_data()["default_admin_password"]
+        admin_password = initial_settings.get("admin_password", default_admin_password)
+        return [
+            "wp",
+            "core",
+            "install",
+            "--url=localhost",
+            "--title=The {} Blog".format(self.model.config["blog_hostname"] or self.app.name),
```

Comment:
Would an f-string work here and be easier to read? The same for the other `.format` calls below

Reply:
Yes, the f-string was introduced in Python 3.6, I think it can be used, I will start update them all, thanks!